cmd/ocm/create/cluster/cmd.go

```diff
@@ -421,12 +500,28 @@ func preRun(cmd *cobra.Command, argv []string) error {
 	// Validate flags / ask for missing data.
 	fs := cmd.Flags()
 
+	// Get options for subscription type
+	subscriptionTypeOptions, _ := getSubscriptionTypeOptions(connection)
+	err = arguments.PromptOneOf(fs, "subscription-type", subscriptionTypeOptions)
+	if err != nil {
+		return err
+	}
+
 	// Only offer the 2 providers known to support OSD now;
 	// but don't validate if set, to not block `ocm` CLI from creating clusters on future providers.
 	providers, _ := osdProviderOptions(connection)
-	err = arguments.PromptOneOf(fs, "provider", providers)
-	if err != nil {
-		return err
+	// If marketplace-gcp subscription type is used, provider can only be GCP
+	gcpBillingModel, _ := getBillingModel(connection, marketplaceGcpSubscriptionType)
+	isGcpSubscriptionType := args.subscriptionType == fmt.Sprintf("%s (%s)", gcpBillingModel.Id(), gcpBillingModel.Description())
```

Comment:
`isGcpSubscriptionType` not all GCP subscriptions falls into this. I would suggest naming it `isGcpMarketplace` instead.

Reply:
Changed